pkg/gitparse/gitparse.go

```diff
@@ -105,6 +105,12 @@ type Diff struct {
 	LineStart int
 	IsBinary  bool
 
+	// Commit metadata.
+	CommitHash    string
```

Comment:
What about keeping these in `Commit` and just having a pointer in `Diff`? That way we don't need to copy all this data for each `Diff`.

Reply:
I defs thought about that, and now that you bring it up, I think i'll go that route.

Reply:
Well actually now that i'm looking at it again, we don't even need the author or date in the commit and longer. We only need the hash to stay in the commit. Is it worth adding a ptr to the commit for just the hash?

Reply:
nvm, using the commit is a good idea after all. 🤣